### PR TITLE
Update 22100 feature list.md

### PR DESCRIPTION
**Pull request by tonybalandiuk**
Took into account the feedback over at https://github.com/o3de/o3de/discussions/12062 in an attempt to highlight some key features for the marketing team.

Signed-off-by: Tony Balandiuk <85971417+tonybalandiuk@users.noreply.github.com>

### CODE REVIEW

File: releases/22.10.0/22100 feature list.md [hunk 1]
```diff
@@ -27,9 +29,11 @@ The general theme of this release is "TBD".
...
 * Script Canvas Architecture update that provides a framework to embed ScriptCanvas functionality in places other than Entity / Component system via the ScriptCanvas Component. The new architecture introduces a set of classes that handle a small part of the ScriptCanvas runtime. The highest level class is the ScriptCanvas::Executor which is now used by the ScriptCanvas::RuntimeComponent. The ScriptCanvas::Executor is also used by the new ScriptCanvasEditor::Interpreter, which is a class that allows developers to embed user access to ScriptCanvas functionality any where in the editor.
 * Writing C++ Script Canvas nodes is now a lot easier! We have deprecated node generics (a set of C++ macros) in favor of using AzAutoGen to produce libraries of functions or standalone nodes. We have consolidated the autogen semantics for Script Canvas grammar nodes and nodeables. We removed the concept of Script Canvas node libraries, anytime you write a node it gets registered automatically. It is no longer necessary to manually reflect or register Script Canvas nodes.
+* Prefab Edit Mode: Now when you double-click on a prefab you now go into prefab edit mode.
```
Comment (hultonha) [inline releases/22.10.0/22100 feature list.md:32]: update: This was technically in the previous release, we just didn't have great feedback. I'd suggest this instead be:

```
* Prefab Edit Mode Visual Feedback: Now when you double-click on a prefab and enter prefab edit mode, there is the option to enable an effect to grey out all entities not inside the current prefab.
```

File: releases/22.10.0/22100 feature list.md [hunk 2]
```diff
@@ -27,9 +29,11 @@ The general theme of this release is "TBD".
...
 * The O3DE editor viewport now comes with a new effect to help show both selected entities (using an outline effect) as well as helping you know what Prefab you're editing by applying a screen effect to everything not in focus. ([Editor Mode Visual Feedback Improvements (Prefab Edit Mode) #3458](https://github.com/o3de/o3de/issues/3458)).
```
Comment (hultonha) [inline releases/22.10.0/22100 feature list.md:33]: update: As we're not calling out the Prefab Edit Mode functionality explicitly above, we can update this line to not include `as well as helping you know what Prefab you're editing by applying a screen effect to everything not in focus`. So this just becomes

```
The O3DE editor viewport now comes with a new feature to help show selected entities with an outline effect.
```

**Review by hultonha — APPROVED** [review]
Small suggestion about 'Prefab Edit Mode' but otherwise looks good (CC @rainbj and @tonybalandiuk)

File: releases/22.10.0/22100 feature list.md (hunk 1 above)
Comment (tonybalandiuk) [reply releases/22.10.0/22100 feature list.md:32]: thanks for the feedback. updated this with your suggestion

File: releases/22.10.0/22100 feature list.md (hunk 2 above)
Comment (tonybalandiuk) [reply releases/22.10.0/22100 feature list.md:33]: thanks. Updated with this proposed text.

**Review by hultonha — APPROVED** [review]
Looks great! 🥳

File: releases/22.10.0/22100 feature list.md (hunk 1 above)
Comment (rainbj) [reply releases/22.10.0/22100 feature list.md]: For future context @hultonha, My hope would be that we can implement double click to go into prefab edit mode in both the viewport and in the outliner. Last tested, this behavior only worked in the outliner. Not sure if we should make a ticket for this part as post release work. @AMZN-daimini let me know if I got that wrong.

**Pull request merged**